sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -87,6 +87,7 @@ class Analyzer(
       EliminateUnions),
     Batch("Resolution", fixedPoint,
       ResolveRelations ::
+      SubstituteHints ::
```

Comment:
Yep. Here is the error when we move it from the batch `Resolution` into the batch `Nondeterministic`.
```
unresolved operator 'Project [*];
org.apache.spark.sql.AnalysisException: unresolved operator 'Project [*];
	at org.apache.spark.sql.catalyst.analysis.CheckAnalysis$class.failAnalysis(CheckAnalysis.scala:40)
	at org.apache.spark.sql.catalyst.analysis.Analyzer.failAnalysis(Analyzer.scala:58)
```